modules/axdriver/src/dyn_drivers/blk/virtio.rs

```
@@ -0,0 +1,112 @@
+extern crate alloc;
+
+use alloc::format;
+use axdriver_base::DeviceType;
+use axdriver_block::BlockDriverOps;
+use axdriver_virtio::MmioTransport;
+use axhal::mem::{MemoryAddr, PhysAddr};
+use core::ptr::NonNull;
+use rdrive::{
+    DriverGeneric, PlatformDevice, driver::block::*, module_driver, probe::OnProbeError,
+    register::FdtInfo,
+};
+
+use crate::dyn_drivers::blk::deverror_to_io_error;
+use crate::dyn_drivers::iomap;
+use crate::virtio::VirtIoHalImpl;
+
+type Device<T> = axdriver_virtio::VirtIoBlkDev<VirtIoHalImpl, T>;
+
+module_driver!(
+    name: "Virtio Block",
+    level: ProbeLevel::PostKernel,
+    priority: ProbePriority::DEFAULT,
+    probe_kinds: &[
+        ProbeKind::Fdt {
+            compatibles: &["virtio,mmio"],
+            on_probe: probe
+        }
+    ],
+);
+
+fn probe(info: FdtInfo<'_>, plat_dev: PlatformDevice) -> Result<(), OnProbeError> {
+    let mut reg = info.node.reg().ok_or(OnProbeError::other(alloc::format!(
+        "[{}] has no reg",
+        info.node.name()
+    )))?;
+
+    let base_reg = reg.next().unwrap();
```

Can we return an error instead of `unwrap`?